openpype/plugins/publish/integrate_inputlinks_ayon.py

```diff
@@ -0,0 +1,160 @@
+import collections
+
+import pyblish.api
+from ayon_api import create_link, make_sure_link_type_exists
+
+from openpype import AYON_SERVER_ENABLED
+
+
+class IntegrateInputLinksAYON(pyblish.api.ContextPlugin):
+    """Connecting version level dependency links"""
+
+    order = pyblish.api.IntegratorOrder + 0.2
+    label = "Connect Dependency InputLinks AYON"
+
+    def process(self, context):
+        """Connect dependency links for all instances, globally
+
+        Code steps:
+        - filter instances that integrated version
+            - have "versionEntity" entry in data
+        - separate workfile instance within filtered instances
+        - when workfile instance is available:
+            - link all `loadedVersions` as input of the workfile
+            - link workfile as input of all other integrated versions
+        - link version's inputs if it's instance have "inputVersions" entry
+        -
+
+        inputVersions:
+            The "inputVersions" in instance.data should be a list of
+            version ids (str), which are the dependencies of the publishing
+            instance that should be extracted from working scene by the DCC
+            specific publish plugin.
+        """
+
+        if not AYON_SERVER_ENABLED:
+            self.log.info("Skipping, not in AYON mode")
+            return
```

Comment:
I know it might be a bit of a hack. But if at the end of this python file you'd do this:
```python
if not AYON_SERVER_ENABLED:
    del IntegrateInputLinksAYON
```
Then it wouldn't even show up in non-AYON mode.
We could do the same to the other integrator. Then at least you wouldn't be seeing the "other" plug-in in the UI and get confused?

(Plus processing is faster if less plug-ins hehe).

What do we think?

Reply:
Will do separate PR.